src/primitives/prp.rs

```diff
@@ -7,6 +7,18 @@ pub struct KnuthShufflePRP<T, const N: usize> {
     permutation: Vec<T>,
 }
 
+fn const_position(input: &[u8], val: u8) -> Option<usize> {
+    let mut position = None;
+
+    for (index, elem) in input.iter().enumerate() {
+        if *elem == val && position == None {
+            position = Some(index);
+        }
+    }
```

Comment:
How can we tell the compiler that we don't want that loop to be short circuited, ever? A smart compiler might undo our intent.

Reply:
Yeah I reckon LLVM is going to fight this pretty hard. We might need to look at doing it in assembly.

Reply:
There is an example approach here: https://github.com/isislovecruft/subtle/blob/master/src/lib.rs